appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/pool/ConnectionPool.java

```diff
@@ -946,7 +946,11 @@ public void resourceClosed(ResourceHandle handle) throws IllegalStateException {
         }
 
         if (!state.isBusy()) {
-            throw new IllegalStateException("state.isBusy() : false");
+            // Do not throw exception, the current transaction should not fail if the state is already 'free'.
+            // This method is for example also called from putbackBadResourceToPool, which also
+            // does not throw the IllegalStateException anymore.
+            // throw new IllegalStateException("state.isBusy() : false");
```

Comment:
Just ... don't leave the dead code just commented out.

Reply:
Thank you, I will do that next time. 
I tried to keep it equivalent to "void resourceErrorOccurred" where the exception was also commented out in the past.

Reply:
I'm not sure what the protocol is, should I clean it up and make a new pull request?

Reply:
You can simply do an extra commit to the same branch as from where you did the PR. That commit will then be added to this PR and all will be fine ;)